src/mon/OSDMonitor.cc

```diff
@@ -8064,6 +8037,40 @@ int OSDMonitor::prepare_new_pool(string& name,
     dout(10) << "prepare_pool_size returns " << r << dendl;
     return r;
   }
+
+  if (pg_num == 0) {
+    auto pg_num_from_mode =
+      [pg_num=g_conf().get_val<uint64_t>("osd_pool_default_pg_num")]
+      (const string& mode,
+       const unsigned calc_pg_num) {
+      return mode == "on" ? (calc_pg_num < pg_num ? calc_pg_num : pg_num) : pg_num;
```

Comment:
`std::min` to make this slightly more readable.

Reply:
No code changes needed as we are skipping this optimization.